server/utils.go

```diff
@@ -174,3 +185,196 @@ func (w *withCode) Format(s fmt.State, verb rune) {
 		}
 	}
 }
+
+func fetchListOfResources(clusterType string, environment string, loginURL string, tenantId string) (map[string][]map[string]map[string]oputil.DataAction, error) {
```

Comment:
can you add a ut for this complicated function?

Reply:
this function is complicated enough so that it probably warrants an comment explaining what it does and what the outcome is